hudi-utilities/src/main/java/org/apache/hudi/utilities/sources/SnapshotLoadQuerySplitter.java

```diff
@@ -61,20 +62,21 @@ public SnapshotLoadQuerySplitter(TypedProperties properties) {
    *
    * @param df The dataset to process.
    * @param beginCheckpointStr The starting checkpoint string.
+   * @param sourceProfileSupplier An Option of a SourceProfileSupplier to use in load splitting implementation
```

Comment:
Let's mark this class with `@PublicAPIClass(maturity = ApiMaturityLevel.EVOLVING)` and the abstract methods with `@PublicAPIMethod(maturity = ApiMaturityLevel.EVOLVING)`, given this class serves as an extendable API class for user to plug in custom implementation.